modules/new_serverpod_auth/serverpod_auth_idp/serverpod_auth_idp_server/test/passkey/passkey_accounts_test.dart

```diff
@@ -0,0 +1,173 @@
+              ),
+            )
+            .build();
+
+        await PasskeyChallenge.db.insertRow(
+          session,
+          PasskeyChallenge(
+            id: challengeId,
+            challenge: _registrationChallenge,
+          ),
+        );
+
+        PasskeyAccounts.config = PasskeyAccountConfig(
+          hostname: 'localhost',
+        );
+      });
+
+      test(
+          "when calling `PasskeyAccounts.registerPasskey`, then the passkey is registered for the session's user.",
+          () async {
+        await PasskeyAccounts.registerPasskey(
+          session,
+          request: PasskeyRegistrationRequest(
+            challengeId: challengeId,
+            keyId: _keyId,
+            publicKey: _publicKey,
+            publicKeyAlgorithm: _publicKeyAlgorithm,
+            clientDataJSON: _registrationClientDataJSON,
+            attestationObject: _attestationObject,
+            authenticatorData: _registrationAuthenticatorData,
+          ),
+        );
+
+        final passkeyAccount = (await PasskeyAccount.db.find(session)).single;
+
+        expect(passkeyAccount.authUserId, user.id);
+      });
+    },
+  );
+
+  withServerpod(
+    'Given an `AuthUser` with a registered passkey,',
+    (final sessionBuilder, final _) {
+      late AuthUserModel user;
+      late Session session;
+
+      final loginChallengeId = const Uuid().v4obj();
+
+      setUp(() async {
+        session = sessionBuilder.build();
+        user = await AuthUsers.create(session);
+
+        PasskeyAccounts.config = PasskeyAccountConfig(
+          hostname: 'localhost',
+        );
+
+        {
+          final registrationChallengeId = const Uuid().v4obj();
+          await PasskeyChallenge.db.insertRow(
+            session,
+            PasskeyChallenge(
+              id: registrationChallengeId,
+              challenge: _registrationChallenge,
+            ),
+          );
+
+          await PasskeyAccounts.registerPasskey(
+            sessionBuilder
+                .copyWith(
+                  authentication: AuthenticationOverride.authenticationInfo(
+                    user.id.uuid,
+                    {},
+                  ),
+                )
+                .build(),
+            request: PasskeyRegistrationRequest(
+              challengeId: registrationChallengeId,
+              keyId: _keyId,
+              publicKey: _publicKey,
+              publicKeyAlgorithm: _publicKeyAlgorithm,
+              clientDataJSON: _registrationClientDataJSON,
+              attestationObject: _attestationObject,
+              authenticatorData: _registrationAuthenticatorData,
+            ),
+          );
+        }
+
+        await PasskeyChallenge.db.insertRow(
+          session,
+          PasskeyChallenge(
+            id: loginChallengeId,
+            challenge: _loginChallenge,
+          ),
+        );
+      });
+
+      test(
+          "when calling `PasskeyAccounts.authenticate`, then the user's ID is returned.",
```

Comment:
Would be good to also have a test for when the user attempts to authenticate with incorrect data on each of the `PasskeyLoginRequest` fields. This one could then be:

```suggestion
          "when calling `PasskeyAccounts.authenticate` with valid login request data, then the user's ID is returned.",
```

Reply:
added the suggested additional cases in 05556009c6fa1344490ee1f8804ce9f11dc307d4